source/developers-guide/legacy-plugin-system/index.md

```diff
@@ -0,0 +1,820 @@
+    ]);
+}
+```
+
+## Add console commands
+For register a console command create a command file which extends the class "ShopwareCommand".
+The register the event `Shopware_Console_Add_Command` and return a new ArrayCollection with your created commands.
+
+```php
+$this->subscribeEvent(
+    'Shopware_Console_Add_Command',
+    'onAddConsoleCommand'
+);
+
+/**
+ * Adds the console commands
+ *
+ * @return ArrayCollection
+ */
+public function onAddConsoleCommand()
+{
+    return new ArrayCollection(
+        [
+          new Command1(),
+          new Command2(),
+        ]
+    );
+}
+```
+
+```php
+<?php
+
+class ExportCommand extends ShopwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        
+    }
+    
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        
+    }
+}
+```
+
+## Add backend emotion components
+For creating custom shopping world elements Shopware provides some helper functions which can be used in the Bootstrap.php of a Shopware plugin. So all you have to do is to create a simple plugin where you can register one ore more elements via the createEmotionComponent() method. As an example for this tutorial we will create a Vimeo element for adding videos to the shopping world.
+ 
+```php
+$vimeoElement = $this->createEmotionComponent([
+    'name' => 'Vimeo Video',
+    'xtype' => 'emotion-components-vimeo',
+    'template' => 'emotion_vimeo',
+    'cls' => 'emotion-vimeo-element',
+    'description' => 'A simple vimeo video element for the shopping worlds.'
+]);
+```
+In the install() method of our plugin we register a new element and save it in the variable $vimeoElement for later use. The createEmotionComponent() method expects a configuration array with the following properties:
+
+### Adding configuration fields to the element
+After registering the new element we can add different form fields to the element which can be filled by the user to configure the element. For each type of field there is a helper function which can be called on the newly registered component. We will add some configuration fields to our example element for the different embed options the Vimeo platform offers.
+
+```php
+$vimeoElement->createTextField([
+    'name' => 'vimeo_video_id',
+    'fieldLabel' => 'Video ID',
+    'supportText' => 'Enter the ID of the video you want to embed.',
+    'allowBlank' => false
+]);
+
+$vimeoElement->createHiddenField([
+    'name' => 'vimeo_video_thumbnail'
+]);
+
+$vimeoElement->createTextField([
+    'name' => 'vimeo_interface_color',
+    'fieldLabel' => 'Interface Color',
+    'supportText' => 'Enter the #hex color code for the video player interface.',
+    'defaultValue' => '#0096FF'
+]);
+
+$vimeoElement->createCheckboxField([
+    'name' => 'vimeo_autoplay',
+    'fieldLabel' => 'Autoplay',
+    'defaultValue' => false
+]);
+
+$vimeoElement->createCheckboxField([
```

Comment:
only show one example for each field type
5x createCheckboxField is useless

Reply:
no.

Reply:
äh doch?

Reply:
D